### PR TITLE
Add use/see_items_while_riding

### PR DESCRIPTION
**Pull request by SFort**
> unascribed/Fabrication-features#23

It felt wierd being able to see the items but not use them so also added that as a separate feature. (wierdly enough turns out it's all done client-side)

### CODE REVIEW

File: features.txt [hunk 1]
```diff
@@ -313,6 +313,13 @@ fixes.silverfish_step
 		possible it's not an oversight and that Mojang simply deemed the silverfish step sound too
 		horrible and dummied it out.
 
+fixes.see_items_while_riding
```
Comment (exaskye) [inline features.txt]: I don't think this belongs under "fixes", as it's a very intentional visual effect. Tweaks is possibly a better fit.

File: features.txt [hunk 2]
```diff
@@ -1231,6 +1238,13 @@ balance.ender_dragon_always_spawn_egg
 	desc:
 		Ender dragon always spawn a dragon egg when killed.
 
+balance.use_items_while_riding
```
Comment (exaskye) [inline features.txt]: Balance feels like a weird place for something this minor, but you're correct in thinking it's a bit much for Tweaks. Weird Tweaks is probably the best fit.

File: src/main/java/com/unascribed/fabrication/mixin/f_balance/use_items_while_riding/MixinMinecraftClient.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayerEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(MinecraftClient.class)
+@EligibleIf(configEnabled="*.use_items_while_riding", envMatches=Env.CLIENT)
+public class MixinMinecraftClient {
+	@Redirect(method="doAttack()V", at=@At(value="INVOKE", target="Lnet/minecraft/client/network/ClientPlayerEntity;isRiding()Z"))
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/f_balance/use_items_while_riding/MixinMinecraftClient.java]: A `Redirect` here makes me a little uneasy. Could this be done with a ModifyVariable?

File: src/main/java/com/unascribed/fabrication/mixin/f_balance/use_items_while_riding/MixinMinecraftClient.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	public boolean canAttack(ClientPlayerEntity clientPlayerEntity){
+		return !MixinConfigPlugin.isEnabled("*.use_items_while_riding") && clientPlayerEntity.isRiding();
+	}
+	@Redirect(method= "doItemUse()V", at=@At(value="INVOKE", target="Lnet/minecraft/client/network/ClientPlayerEntity;isRiding()Z"))
+	public boolean canUse(ClientPlayerEntity clientPlayerEntity){
+		return !MixinConfigPlugin.isEnabled("*.use_items_while_riding") && clientPlayerEntity.isRiding();
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/f_balance/use_items_while_riding/MixinMinecraftClient.java]: Ditto.

File: src/main/java/com/unascribed/fabrication/mixin/b_utility/see_items_while_riding/MixinHeldItemRenderer.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.client.render.item.HeldItemRenderer;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Mixin(HeldItemRenderer.class)
+@EligibleIf(anyConfigEnabled={"*.see_items_while_riding", "*.use_items_while_riding"}, envMatches=Env.CLIENT)
+public class MixinHeldItemRenderer {
+	@Redirect(method="updateHeldItems()V", at=@At(value="INVOKE", target="Lnet/minecraft/client/network/ClientPlayerEntity;isRiding()Z"))
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/b_utility/see_items_while_riding/MixinHeldItemRenderer.java]: Ditto.

File: src/main/java/com/unascribed/fabrication/mixin/f_balance/use_items_while_riding/MixinMinecraftClient.java (hunk 3 above)
Comment (SFort) [reply src/main/java/com/unascribed/fabrication/mixin/f_balance/use_items_while_riding/MixinMinecraftClient.java]: While i'm assuming this is done with At("JUMP"). i can't figure out how.

**Pull request merged**